### PR TITLE
Add a configuration parameter to use entry_posted journal setting

### PR DESCRIPTION
**Pull request by laetitia-gangloff**

### CODE REVIEW

File: account_default_draft_move/account_bank_statement.py [hunk 1]
```diff
@@ -27,5 +27,11 @@ class AccountBankStatement(models.Model):
     def button_confirm_bank(self):
         res = super(AccountBankStatement, self).button_confirm_bank()
         entries = self.mapped('line_ids.journal_entry_id')
+        use_journal_setting = bool(self.env['ir.config_parameter'].get_param(
+            'use_journal_setting', default=False))
+        if use_journal_setting:
+            for move in entries:
+                if move.journal_id.entry_posted:
+                    entries.remove(move)
```
Comment (sbidoul) [inline account_default_draft_move/account_bank_statement.py]: removing items from a list while iterating it is a bit delicate.

Why not `if use_journal_settings: entries = [ e for e in entries if not e.journal_id_entry_posted ]`

File: account_default_draft_move/account.py [hunk 2]
```diff
@@ -27,8 +27,12 @@ class AccountInvoice(models.Model):
     def action_move_create(self):
         """Set move line in draft state after creating them."""
         res = super(AccountInvoice, self).action_move_create()
+        use_journal_setting = bool(self.env['ir.config_parameter'].get_param(
+            'use_journal_setting', default=False))
```
Comment (JonathanNEMRY) [inline account_default_draft_move/account.py:31]: I think that the default value for `default` is `False`?

File: account_default_draft_move/account_bank_statement.py [hunk 3]
```diff
@@ -27,5 +27,10 @@ class AccountBankStatement(models.Model):
     def button_confirm_bank(self):
         res = super(AccountBankStatement, self).button_confirm_bank()
         entries = self.mapped('line_ids.journal_entry_id')
+        use_journal_setting = bool(self.env['ir.config_parameter'].get_param(
+            'use_journal_setting', default=False))
```
Comment (JonathanNEMRY) [inline account_default_draft_move/account_bank_statement.py:31]: Same comment here

File: account_default_draft_move/account.py (hunk 2 above)
Comment (sbidoul) [inline account_default_draft_move/account.py:31]: use_journal_settings is perhaps difficult to understand when you look at the list of system parameters. The name of the parameter could be more explicit.

**Pull request merged**